types/stripe/index.d.ts

```diff
@@ -4505,7 +4532,8 @@ declare namespace Stripe {
              */
             source:
                 | string
-                | IStripeSource;
+                | IStripeSource
+                | null;
```

Comment:
This will be null if the PaymentIntent is used with a PaymentMethod instead of a source.